### PR TITLE
Add OpenAPI schema

### PR DESCRIPTION
**Pull request by plojyon**

### CODE REVIEW

File: resolwe/flow/views/mixins.py [hunk 1]
```diff
@@ -149,7 +149,9 @@ def get_object_with_lock(self):
 class ResolweCheckSlugMixin:
     """Slug validation."""
 
-    @extend_schema(responses={200: bool, 401: None})
+    @extend_schema(
+        parameters=[OpenApiParameter("name", str)], responses={status.HTTP_200_OK: bool}
```
Comment (gregorjerse) [inline resolwe/flow/views/mixins.py]: How about 400 and 401?

**Review by gregorjerse — CHANGES_REQUESTED** [review]

File: resolwe/flow/views/mixins.py [hunk 2]
```diff
@@ -147,6 +149,7 @@ def get_object_with_lock(self):
 class ResolweCheckSlugMixin:
     """Slug validation."""
 
+    @extend_schema(responses={200: bool, 401: None})
```
[the remark this answers is not in the document]
Comment (plojyon) [reply resolwe/flow/views/mixins.py]: I try not to document 4xx and 5xx errors for simplicity. This was amended in a later commit, but wasn't squashed.

File: resolwe/permissions/mixins.py [hunk 3]
```diff
@@ -28,6 +30,22 @@ class CurrentUserPermissionsSerializer(serializers.Serializer):
     permissions = serializers.ListField(child=serializers.CharField(max_length=30))
 
 
+class PermissionSerializer(serializers.Serializer):
+    """Permission serializer."""
+
+    permission_choices = [Permission(i).name for i in list(Permission.highest())]
+
+    users = serializers.DictField(
```
[the remark this answers is not in the document]
Comment (plojyon) [reply resolwe/permissions/mixins.py:39]: Dict keys cannot be integers, unfortunately.

File: resolwe/flow/filters.py [hunk 4]
```diff
@@ -544,7 +545,7 @@ def filter_for_field(cls, field, field_name, lookup_expr=None):
         filter that checks the permissions on the collections.
         """
 
-        def collection_permission_filter(self, qs, value):
+        def collection_permission_filter(self, qs, value: str):
```
Comment (gregorjerse) [inline resolwe/flow/filters.py:548]: Add type to the queryset (argument qs).

File: resolwe/flow/filters.py [hunk 5]
```diff
@@ -616,7 +617,7 @@ class AnnotationValueMetaclass(ResolweFilterMetaclass):
     def __new__(mcs, name, bases, namespace):
         """Inject extensions into the filter."""
 
-        def filter_permissions(self, qs, value, original_filter):
+        def filter_permissions(self, qs, value: str, original_filter):
```
Comment (gregorjerse) [inline resolwe/flow/filters.py:620]: Add type hint to the queryset. You could also type `original_filter`: it is a callable that accepts a queryset and a value and returns a new queryset.

File: resolwe/permissions/mixins.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (gregorjerse) [reply resolwe/permissions/mixins.py:39]: Can we at least add a comment they represent username/group ?

**Review by gregorjerse — CHANGES_REQUESTED** [review]

**Review by gregorjerse — APPROVED** [review]

**Pull request merged**